javaslang/src/main/java/javaslang/collection/TreeSet.java

```diff
@@ -505,9 +505,10 @@
         }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public boolean contains(T element) {
-        return tree.contains(element);
+    public boolean contains(Object element) {
+        return tree.contains((T) element);
```

Comment:
I'm not sure about that...
This line can produce <code>ClassCastException</code> if <code>Comparator</code> do not check this.